airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -164,6 +165,9 @@ def get_dag_runs(
     else:
         query = query.filter(DagRun.dag_id == dag_id)
 
+    if state:
+        query = query.filter(DagRun.state == state)
```

Comment:
```suggestion
    if state:
        query = query.filter(DagRun.state.in_(state))
```
The `FilterState` query component is an array.